File: pfib.nim
import threadpool
{.experimental.}

# compile with:
#   nim c --threads:on -d:release pfib.nim

proc fib(n: int): int {.noinit.} =
  if n < 3:
    1
  else:
    fib(n-1) + fib(n-2)

proc echofib(n :int) =
  echo "fib(", n, ") = ", fib(n)

parallel:
  for i in 44 .. 46:
    spawn echofib(i)

File: pfib.cs
// compile: mcs pfib.cs
// run    : mono pfib.exe

using System;
using System.Threading.Tasks;

class Fibber
{
  static int fib(int n)
  {
    return (n < 3) ? 1 : fib(n-2) + fib(n-1);
  }

  static void Main()
  {
    Parallel.For(44, 47, i =>
    {
      Console.WriteLine("fib({0}) = {1}", i, fib(i));
    });
  }
}
